### PR TITLE
add findOnly/Without methods

### PR DESCRIPTION
**Pull request by Nevrdid**
convert all existing find compatibles.

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
[democratic collaboration](https://github.com/TooAngel/democratic-collaboration)
Approved reviews will speed up the merge, request changes will slow it down.

Please review the PR to help.

Summoning some reviewers:
 - @TooAngel: Some warm words would be nice
 - @Somotaw: Not sure why you are here
 - @WhiteHalmos: A proper review is appreciated

File: src/prototype_room_utils.js [hunk 1]
```diff
@@ -22,6 +22,22 @@ Room.prototype.nearestRoomName = function(roomsNames, limit) {
...
 
+Room.prototype.findWithoutStructType = function(findTarget, excludes) {
+  let table = {};
+  _.each(excludes, e => table[e] = true);
+  return this.find(findTarget, {
+    filter: s => !table[s.structureType]
+  });
+};
+
+Room.prototype.findOnlyStructType = function(findTarget, excludes) {
```
Comment (TooAngel) [inline src/prototype_room_utils.js]: I think `excludes` should be `includes`

**Comment by democratic-collaboration** [comment]
DCBOT: A new review, yeah.

    Votes: 208/318
    Coefficient: 0.654088050314
    Merging in 2 days 23 hours
    Age 0 days 11 hours

File: src/prototype_room_utils.js (hunk 1 above)
Comment (Nevrdid) [inline src/prototype_room_utils.js]: 👍

File: src/prototype_creep_resources.js [hunk 2]
```diff
@@ -627,9 +619,7 @@ Creep.prototype.construct = function() {
       for (let structureId in structures) {
         let structure = structures[structureId];
         if (structure.structureType === STRUCTURE_SPAWN) {
-          let spawns = this.room.find(FIND_STRUCTURES, {
-            filter: filterSpawns
```
Comment (Nevrdid) [inline src/prototype_creep_resources.js]: 👍

**Pull request merged**